hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/PrngSystemContract.java

```diff
@@ -108,38 +119,72 @@ void createSuccessfulRecord(
             requireNonNull(randomNum);
             requireNonNull(contractID);
             var updater = (ProxyWorldUpdater) frame.getWorldUpdater();
-            updater.externalizeSystemContractResults(
-                    contractFunctionResultSuccessFor(gasRequirement, randomNum, contractID), SUCCESS);
+            final var senderId = ((ProxyEvmAccount) updater.getAccount(frame.getSenderAddress())).hederaId();
+
+            var data = contractFunctionResultSuccessFor(
+                    gasRequirement, randomNum, frame.getRemainingGas(), frame.getInputData(), senderId);
+
+            updater.enhancement()
+                    .systemOperations()
+                    .dispatch(
+                            synthBody(),
+                            new ActiveContractVerificationStrategy(
```

Comment:
This `UtilPrngTransactionBody` has no non-payer signatures required.

So really we could pass in a `VerificationStrategy` that just always returns `Decision.INVALID`, since it will never be used.

Just a nice-to-have, not blocking.